zebra-rpc/src/config/mining.rs

```diff
@@ -29,6 +29,15 @@ pub struct Config {
     ///
     /// This developer-only config is not supported for general use.
     pub debug_like_zcashd: bool,
+
+    /// Mine blocks using Zebra's internal miner, without an external mining pool or equihash solver.
+    ///
+    /// This experimental feature is only supported on testnet.
+    /// Mainnet miners should use a mining pool with GPUs or ASICs designed for efficient mining.
```

Comment:
I can't see where in the code we are not allowing the miner to run in mainnet.

I also think the text should say the feature is supported in public testnet and custom testnets ?

Reply:
> I can't see where in the code we are not allowing the miner to run in mainnet.

It's not explicitly disallowed anywhere, it just wouldn't work very well with the CPU equihash solver and won't work at all until that's restored.

> I also think the text should say the feature is supported in public testnet and custom testnets ?

Updated the docs in https://github.com/ZcashFoundation/zebra/pull/8506/commits/40e6081dfa3aafffed2d889d000a098117d43821 to say that it's currently only supported on Regtest since it's currently the only one that disables proof of work validation.

Reply:
do you think we should panic if the network is not regtest ?

Reply:
It wouldn't hurt to log a warning and skip spawning the miner on unsupported networks (if `disable_pow` is false) until the equihash solver is restored, but I don't think it's necessary for an experimental feature.

Once the equihash solver is restored, CPU mining on Mainnet still isn't worthwhile, but I see no harm in allowing it.